site/tools/manifests/user/managed.pp

```diff
@@ -123,6 +124,17 @@
     create_resources(ssh_authorized_key, $sshkeys_content, $sshkey_defaults)
   }
 
+  if $generate_ssh_keypair {
+    if $id_rsa_source == '' and $id_rsa_pub_source == '' {
+      tools::ssh_keygen { $name:
```

Comment:
FYI. I don't know if this comes for a change of yours or something already in old code somehow shown here, in any case note that both the old profile and the tools modules when migrated to the single psick module. Defines were renamed from tools::ssh_keygen to, in this case, psick::ssh::keygen.